src/firebase_functions/options.py

```diff
@@ -222,7 +234,27 @@ def _asdict_with_global_options(self) -> dict:
             merged_options["labels"] = {**_GLOBAL_OPTIONS.labels, **self.labels}
         if "labels" not in merged_options:
             merged_options["labels"] = {}
-
+        preserve_external_changes: bool = merged_options.get(
+            "preserve_external_changes",
+            False,
+        )
+        resettable_options = [
+            "memory",
+            "timeout_sec",
+            "min_instances",
+            "max_instances",
+            "ingress",
+            "concurrency",
+            "service_account",
+            "vpc_connector",
+            "vpc_connector_egress_settings",
+        ]
+        if not preserve_external_changes:
+            for option in resettable_options:
+                if option not in merged_options:
+                    merged_options[option] = RESET_VALUE
+        if "preserve_external_changes" in merged_options:
+            del merged_options["preserve_external_changes"]
```

Comment:
q: why is this logic necessary

Reply:
It's not needed, have pushed a change to remove it now